File: tests/gen.swift
import Foundation;import CommonCrypto
let p="parity_images/swift_database.ppm";let d="P6\n4000 4000\n255\n".data(using:.utf8)!+Data(count:48000000)
try? d.write(to:URL(fileURLWithPath:p));let f=FileHandle(forUpdatingAtPath:p)!
for i in 1...4000{let s="Node_Alpha_\(i)ScyWeb_Global_Parity_2026";let data=s.data(using:.utf8)!
var h=[UInt8](repeating:0,count:Int(CC_SHA256_DIGEST_LENGTH))
data.withUnsafeBytes{_ = CC_SHA256($0.baseAddress, CC_LONG(data.count), &h)}
let x=(Int(h[0])<<8|Int(h[1]))%4000;let y=(Int(h[2])<<8|Int(h[3]))%4000
f.seek(toFileOffset:UInt64(17+(y*4000+x)*3));var v="UPDATE user SET status='ACTIVE' WHERE id=\(i)".data(using:.utf8)!
v.append(0);f.write(v)};f.closeFile()

File: tests/gen.rs
use std::io::{Write, Seek, SeekFrom};
use std::fs::OpenOptions;
fn main() {
    let p = "parity_images/rust_database.ppm";
    let mut f = std::fs::File::create(p).unwrap();
    f.write_all(b"P6\n4000 4000\n255\n").unwrap();
    f.set_len(48000017).unwrap();
    let mut f = OpenOptions::new().write(true).open(p).unwrap();
    for i in 1..=4000 {
        let mut hasher = sha2::Sha256::new();
        hasher.update(format!("{}{}{}", "Node_Alpha_", i, "ScyWeb_Global_Parity_2026").as_bytes());
        let h = format!("{:x}", hasher.finalize());
        let x = u64::from_str_radix(&h[0..4], 16).unwrap() % 4000;
        let y = u64::from_str_radix(&h[4..8], 16).unwrap() % 4000;
        let o = 17 + (y * 4000 + x) * 3;
        f.seek(SeekFrom::Start(o)).unwrap();
        f.write_all(format!("UPDATE user SET status='ACTIVE' WHERE id={}\0", i).as_bytes()).unwrap();
    }
}
